### ISSUE TITLE
Airton AC not fully recognised

### ISSUE THREAD

**Issue by ejalal** [rospogrigio/localtuya#1115]
## The problem

Airton AC has some missing data points to finish integration, on/off and temperature change works OK, but I would like at least to be able to change Operation Mode from the UI to Heating/Cooling/Ventilation/Dehumidification. Changing Fan speed or enabling Swing could be a bonus.

HVAC Mode DP (optional) => Set tp DP 4, but always says Auto !
HVAC Mode Set (optional) => Don't know what to put in here but I'm expecting a list for auto, cold, wet, heat, fan
HVAC Current Action DP (optional) => Don't know what to put in here

## Environment

- Localtuya version: ### 4.1.1 
- Last working localtuya version (if known and relevant): 
- Home Assistant Core version: 2022.11.1 
- [] Are you using the Home Assistant Tuya Cloud component ? yes
- [] Are you using the Tuya App in parallel ? yes

## Steps to reproduce

1.
2. 
3.

## Configuration `configuration.yaml` or `config_flow`

```yaml

```

## DP dump

```
    DPS [1] VALUE [True]
    DPS [2] VALUE [240]
    DPS [3] VALUE [250]
    DPS [4] VALUE [heat]
    DPS [5] VALUE [mute]
    DPS [8] VALUE [False]
    DPS [9] VALUE [False]
    DPS [12] VALUE [True]
    DPS [13] VALUE [True]
    DPS [15] VALUE [off]
    DPS [20] VALUE [0]
    DPS [22] VALUE [0]
    DPS [101] VALUE [0]
    DPS [102] VALUE [78]
    DPS [103] VALUE [1]
    DPS [105] VALUE [c]
    DPS [106] VALUE [off]
    DPS [107] VALUE [off]
    DPS [108] VALUE [False]
    DPS [109] VALUE [False]
    DPS [110] VALUE [True]
    DPS [111] VALUE [False]
    DPS [112] VALUE [cold]
    DPS [113] VALUE [0]
    DPS [114] VALUE [heat]
    DPS [115] VALUE [False]
```
## Provide Home Assistant taceback/logs

```txt

```
## Additional information

API call for "Get the specifications and properties of the device"
```
{
  "result": {
    "category": "kt",
    "functions": [
      {
        "code": "switch",
        "desc": "{}",
        "name": "开关",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "eco",
        "desc": "{}",
        "name": "ECO模式",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "drying",
        "desc": "{}",
        "name": "干燥模式",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "cleaning",
        "desc": "{}",
        "name": "自清洁",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "temp_unit_convert",
        "desc": "{\"range\":[\"c\",\"f\"]}",
        "name": "温标,华氏度，摄氏度切换(新)",
        "type": "Enum",
        "values": "{\"range\":[\"c\",\"f\"]}"
      },
      {
        "code": "temp_set",
        "desc": "{\"unit\":\"℃\",\"min\":160,\"max\":880,\"scale\":1,\"step\":10}",
        "name": "设置温度",
        "type": "Integer",
        "values": "{\"unit\":\"℃\",\"min\":160,\"max\":880,\"scale\":1,\"step\":10}"
      },
      {
        "code": "mode",
        "desc": "{\"range\":[\"auto\",\"cold\",\"wet\",\"heat\",\"fan\"]}",
        "name": "工作模式",
        "type": "Enum",
        "values": "{\"range\":[\"auto\",\"cold\",\"wet\",\"heat\",\"fan\"]}"
      },
      {
        "code": "heat",
        "desc": "{}",
        "name": "辅热",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "light",
        "desc": "{}",
        "name": "灯光",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "sleep",
        "desc": "{}",
        "name": "睡眠功能",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "health",
        "desc": "{}",
        "name": "健康",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "fan_speed_enum",
        "desc": "{\"range\":[\"auto\",\"low\",\"low_mid\",\"mid\",\"mid_high\",\"high\",\"mute\",\"turbo\"]}",
        "name": "风速模式",
        "type": "Enum",
        "values": "{\"range\":[\"auto\",\"low\",\"low_mid\",\"mid\",\"mid_high\",\"high\",\"mute\",\"turbo\"]}"
      }
    ],
    "status": [
      {
        "code": "switch",
        "name": "开关",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "fan_speed_enum",
        "name": "风速模式",
        "type": "Enum",
        "values": "{\"range\":[\"auto\",\"low\",\"low_mid\",\"mid\",\"mid_high\",\"high\",\"mute\",\"turbo\"]}"
      },
      {
        "code": "eco",
        "name": "ECO模式",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "drying",
        "name": "干燥模式",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "cleaning",
        "name": "自清洁",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "temp_unit_convert",
        "name": "温标,华氏度，摄氏度切换(新)",
        "type": "Enum",
        "values": "{\"range\":[\"c\",\"f\"]}"
      },
      {
        "code": "temp_set",
        "name": "设置温度",
        "type": "Integer",
        "values": "{\"unit\":\"℃\",\"min\":160,\"max\":880,\"scale\":1,\"step\":10}"
      },
      {
        "code": "temp_current",
        "name": "当前温度",
        "type": "Integer",
        "values": "{\"unit\":\"℃\",\"min\":0,\"max\":600,\"scale\":1,\"step\":1}"
      },
      {
        "code": "mode",
        "name": "工作模式",
        "type": "Enum",
        "values": "{\"range\":[\"auto\",\"cold\",\"wet\",\"heat\",\"fan\"]}"
      },
      {
        "code": "heat",
        "name": "辅热",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "light",
        "name": "灯光",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "countdown_left",
        "name": "倒计时剩余时间",
        "type": "Integer",
        "values": "{\"unit\":\"分钟\",\"min\":0,\"max\":1440,\"scale\":0,\"step\":1}"
      },
      {
        "code": "sleep",
        "name": "睡眠功能",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "health",
        "name": "健康",
        "type": "Boolean",
        "values": "{}"
      }
    ]
  },
  "success": true,
  "t": 1667828509086,
  "tid": "edc965ed5ea111ed8c069ee482573564"
}
```

**Comment by sambarlick** [comment]
I just saw your post as I was about to create a bug report myself with a similar issue.

My Aircon has Heat, Cool, Dry(Dehumidify) and Fan modes. Even though I link the correct DP to the HVAC mode which includes these modes, the options Tuya Local offer are Auto and Heat. This is not good enough and all modes should be supported.

I agree fan speed and swing should be included but I had to create a seperate fan entity within the device to handle the fan speed and swing.

**Comment by balloo42** [comment]
Hi,
I modified the climate.py (/config/custom_components/localtuya/climate.py) to add the missing modes for my Airton HVAC.

I added the following modes : HVAC_MODE_COOL,  HVAC_MODE_DRY, HVAC_MODE_FAN_ONLY 

from homeassistant.components.climate.const import (
    CURRENT_HVAC_HEAT,
    CURRENT_HVAC_IDLE,
    HVAC_MODE_AUTO,
    HVAC_MODE_HEAT,
    HVAC_MODE_OFF,
    HVAC_MODE_COOL,
    HVAC_MODE_DRY,
    HVAC_MODE_FAN_ONLY,
    PRESET_AWAY,
    PRESET_ECO,
    PRESET_HOME,
    PRESET_NONE,
    SUPPORT_PRESET_MODE,
    SUPPORT_TARGET_TEMPERATURE,
    SUPPORT_TARGET_TEMPERATURE_RANGE,
)

And i added the following mode_set : auto/cold/wet/heat/fan 

HVAC_MODE_SETS = {
    "manual/auto": {
        HVAC_MODE_HEAT: "manual",
        HVAC_MODE_AUTO: "auto",
    },
    "Manual/Auto": {
        HVAC_MODE_HEAT: "Manual",
        HVAC_MODE_AUTO: "Auto",
    },
    "Manual/Program": {
        HVAC_MODE_HEAT: "Manual",
        HVAC_MODE_AUTO: "Program",
    },
    "True/False": {
        HVAC_MODE_HEAT: True,
    },
    "1/0": {
        HVAC_MODE_HEAT: "1",
        HVAC_MODE_AUTO: "0",
    },
    "auto/cold/wet/heat/fan": {
        HVAC_MODE_HEAT: "heat",
        HVAC_MODE_AUTO: "auto",
        HVAC_MODE_COOL: "cold",
        HVAC_MODE_DRY: "wet",
        HVAC_MODE_FAN_ONLY: "fan",        
    },    


![image](https://user-images.githubusercontent.com/96748205/202905324-d5ca49ac-9f33-49a4-9d9d-08d2934eb41f.png)

In order to control the fan speed you need to add a select with the following entries : auto;low;low_mid;mid;mid_high;high;mute;turbo

![image](https://user-images.githubusercontent.com/96748205/202906266-906a145e-7d9b-4e86-9e7e-bce5fa12c4f7.png)

**Comment by carmelo42** [comment]
Hello @balloo42 !
Can you provide more information about localTuya configuration please ?

I have modified my climate.py file like you said, but I don't know how to configure the Airton device in local Tuya integration : 

<img width="571" alt="image" src="https://user-images.githubusercontent.com/2070041/208731095-3b11d6ea-4078-4b13-b9f7-e2977bb30212.png">

Any help would be much appreciated :)

**Comment by balloo42** [comment]
Hello @carmelo42,

My settings : 

![image](https://user-images.githubusercontent.com/96748205/208740565-140b3470-f2cb-4424-ace7-3c71443028cc.png)
![image](https://user-images.githubusercontent.com/96748205/208740615-a75f3682-d1f3-4452-9f0f-e7101b99f300.png)

After that screen, i unchecked the "Do not add any more entities", and i chose the entity type "select" to configure the fan speed on DP 5 (values are : auto;low;low_mid;mid;mid_high;high;mute;turbo), i also added a switch for "Health" (DP 110), and "light" (DP 13).

**Comment by carmelo42** [comment]
Thanks @balloo42 it is way better now : I can turn on / off the AC. But when I'm select "heat" the mode automatically switch on "Auto" itself.

If I select cold, or fan only, or dry, the mode stays on the good value.

**Comment by carmelo42** [comment]
Thanks again @balloo42 it's now fully working with Airton - Yuzu - Fujicool HVAC devices (self installables HVAC devices). 
I've forked the repo and added my modifications : https://github.com/carmelo42/localtuya/tree/fujicool-yuzu-hvac

**Comment by balloo42** [comment]
looks like there is a slight difference between our mode_sets : heat vs hot and fan vs wind
happy to share and that you figured this out.

**Comment by PhilippeP62** [comment]
Well I also have a Airton unit so I can say that you both helped me a lot !
Thanks

**Comment by CFTechno** [comment]
Thank you all! That really helped. 

Small question however: does any of you know the DP for "auxiliaryHeat" and/or "8c heat" Specially the last one is interesting because you can set it and it will work. If you set the "8c heat" the unit will beep but no indication is activated in the smart life app but it does work.

**Comment by PhilippeP62** [comment]
AuxiliaryHeat is DP12 
No Idea about 8cHeat but I allready do that in automation.

**Comment by uriirichard** [comment]
> Hi, I modified the climate.py (/config/custom_components/localtuya/climate.py) to add the missing modes for my Airton HVAC.
> 
> I added the following modes : HVAC_MODE_COOL, HVAC_MODE_DRY, HVAC_MODE_FAN_ONLY
> 
> from homeassistant.components.climate.const import ( CURRENT_HVAC_HEAT, CURRENT_HVAC_IDLE, HVAC_MODE_AUTO, HVAC_MODE_HEAT, HVAC_MODE_OFF, HVAC_MODE_COOL, HVAC_MODE_DRY, HVAC_MODE_FAN_ONLY, PRESET_AWAY, PRESET_ECO, PRESET_HOME, PRESET_NONE, SUPPORT_PRESET_MODE, SUPPORT_TARGET_TEMPERATURE, SUPPORT_TARGET_TEMPERATURE_RANGE, )
> 
> And i added the following mode_set : auto/cold/wet/heat/fan
> 
> HVAC_MODE_SETS = { "manual/auto": { HVAC_MODE_HEAT: "manual", HVAC_MODE_AUTO: "auto", }, "Manual/Auto": { HVAC_MODE_HEAT: "Manual", HVAC_MODE_AUTO: "Auto", }, "Manual/Program": { HVAC_MODE_HEAT: "Manual", HVAC_MODE_AUTO: "Program", }, "True/False": { HVAC_MODE_HEAT: True, }, "1/0": { HVAC_MODE_HEAT: "1", HVAC_MODE_AUTO: "0", }, "auto/cold/wet/heat/fan": { HVAC_MODE_HEAT: "heat", HVAC_MODE_AUTO: "auto", HVAC_MODE_COOL: "cold", HVAC_MODE_DRY: "wet", HVAC_MODE_FAN_ONLY: "fan", },
> 
> ![image](https://user-images.githubusercontent.com/96748205/202905324-d5ca49ac-9f33-49a4-9d9d-08d2934eb41f.png)
> 
> In order to control the fan speed you need to add a select with the following entries : auto;low;low_mid;mid;mid_high;high;mute;turbo
> 
> ![image](https://user-images.githubusercontent.com/96748205/202906266-906a145e-7d9b-4e86-9e7e-bce5fa12c4f7.png)

Thanks, with this, and some from https://github.com/rospogrigio/localtuya/pull/888 I finnaly have a PROKLIMA-A018-12KR2/C fully working!

**Comment by PhilippeP62** [comment]
Still missing from 5.0.1 ...

**Comment by sanyajorj** [comment]
climate.py 
line 289  ........... self._current_temperature = self._current_temperature * 10
Change 10 to 1 and the temperature will be displayed correctly

**Comment by patpac9** [comment]
I can't set "HVAC Mode Set" to this...
I only have the following options.
![grafik](https://github.com/rospogrigio/localtuya/assets/103025196/258c1cc0-ffba-4e9a-9a07-fb5f6259a0ef)

**Comment by tolnai** [comment]
I have Polar AC, but ended up here because their DPs seem to be similar (e.g. ["cold", "hot", "wet", "wind", "auto"] mode).
How could we speed up adding handling them to localtuya? Create a separate PR from [this one](https://github.com/rospogrigio/localtuya/pull/1223) only for the HVAC side?

**Comment by OpenGrowBox** [comment]
its also working for Air Condition-A012 from Tuya the Costway 4-1 in Device. But only the Dry mode looks not to work for me. 

all other modes get selected only the Dry not. 

here is the code part

    "auto/cold/wet/heat/fan": {
	HVAC_MODE_HEAT: "Heat",
	HVAC_MODE_AUTO: "Auto",
	HVAC_MODE_COOL: "Cool",
	HVAC_MODE_DRY: "Dry",
	HVAC_MODE_FAN_ONLY: "Fan",
    },

have anybody got the same issue that one mode isnt working ?

**Comment by OpenGrowBox** [comment]
found the issue on the API it self 
on the DEVICE is a TYPO for DRY - There is  a enum with DYR
![image](https://github.com/rospogrigio/localtuya/assets/170749418/b69b7c4e-8ff0-4aea-82a3-6a16bada04c3)

**Comment by bumptronic** [comment]
> its also working for Air Condition-A012 from Tuya the Costway 4-1 in Device. But only the Dry mode looks not to work for me.
> 
> all other modes get selected only the Dry not.
> 
> here is the code part
> 
> ```
> "auto/cold/wet/heat/fan": {
> HVAC_MODE_HEAT: "Heat",
> HVAC_MODE_AUTO: "Auto",
> HVAC_MODE_COOL: "Cool",
> HVAC_MODE_DRY: "Dry",
> HVAC_MODE_FAN_ONLY: "Fan",
> },
> ```
> 
> have anybody got the same issue that one mode isnt working ?

Have you tried:

HVAC_MODE_DRY: "wet",

**Comment by OpenGrowBox** [comment]
no it was a typo on the API from tuya it self it should be "Dry" but it is "Dyr" 

with "Dyr" all works fine :D some times we nned typos in our life :D haha

**Comment by Mastiffen** [comment]
Thank you all for this info! I have an air conditioning called A4 in Tuya, Lyfco is the brand. It looks like this:

![image](https://github.com/rospogrigio/localtuya/assets/35197020/ec20c011-ecc9-4e25-bdbf-c656e1ec0cb5)

I too have the problem that when I set it to cool it shows auto in Lovelace. Is this because it's a unit that isn't added to the system?

**Comment by thepaulf** [comment]
> Hello @carmelo42,
> 
> My settings :
> 
> ![image](https://user-images.githubusercontent.com/96748205/208740565-140b3470-f2cb-4424-ace7-3c71443028cc.png) ![image](https://user-images.githubusercontent.com/96748205/208740615-a75f3682-d1f3-4452-9f0f-e7101b99f300.png)
> 
> After that screen, i unchecked the "Do not add any more entities", and i chose the entity type "select" to configure the fan speed on DP 5 (values are : auto;low;low_mid;mid;mid_high;high;mute;turbo), i also added a switch for "Health" (DP 110), and "light" (DP 13).

What setting do you use for the first ID field (you don't see it in your screenshot because you're editting.)

**Comment by Fullgame90** [comment]
Hello everyone, since updating local tuya (5.2.3) the syntax of the climate.py file has changed and I can't modify it to add the modes of my airton air conditioning (heat and fan) as before (example at the top of the topic). Can anyone help?

**Comment by balloo42** [comment]
Hello @Fullgame90 

In HVAC_MODE_SETS = {

i added : 

    "Auto/Cold/Wet/Heat/Fan": {
        HVACMode.HEAT: "heat",
        HVACMode.FAN_ONLY: "fan",
        HVACMode.DRY: "wet",
        HVACMode.COOL: "cold",
        HVACMode.AUTO: "auto",
    },

I also had to reconfigure/edit the device in LocalTuya. I had to change the "precision" to 0.1 in order to correct the temperature, and i may have changed something else but i don't remember what.


![Image](https://github.com/user-attachments/assets/a26938f9-f258-4986-9720-69c041cec903)

![Image](https://github.com/user-attachments/assets/b959a487-df6e-4d1c-becc-39acaea9c9b7)

![Image](https://github.com/user-attachments/assets/e11ecd4e-4c85-4ac8-9ead-c4ef02431e86)

**Comment by balloo42** [comment]
@Fullgame90 

Here it is : 

HVAC_MODE_SETS = {
    "manual/auto": {
        HVACMode.HEAT: "manual",
        HVACMode.AUTO: "auto",
    },
    "Manual/Auto": {
        HVACMode.HEAT: "Manual",
        HVACMode.AUTO: "Auto",
    },
    "MANUAL/AUTO": {
        HVACMode.HEAT: "MANUAL",
        HVACMode.AUTO: "AUTO",
    },
    "Manual/Program": {
        HVACMode.HEAT: "Manual",
        HVACMode.AUTO: "Program",
    },
    "m/p": {
        HVACMode.HEAT: "m",
        HVACMode.AUTO: "p",
    },
    "True/False": {
        HVACMode.HEAT: True,
    },
    "Auto/Cold/Dry/Wind/Hot": {
        HVACMode.HEAT: "hot",
        HVACMode.FAN_ONLY: "wind",
        HVACMode.DRY: "wet",
        HVACMode.COOL: "cold",
        HVACMode.AUTO: "auto",
    },
    "Auto/Cold/Wet/Heat/Fan": {
        HVACMode.HEAT: "heat",
        HVACMode.FAN_ONLY: "fan",
        HVACMode.DRY: "wet",
        HVACMode.COOL: "cold",
        HVACMode.AUTO: "auto",
    },
    "Cold/Dehumidify/Hot": {
        HVACMode.HEAT: "hot",
        HVACMode.DRY: "dehumidify",
        HVACMode.COOL: "cold",
    },
    "1/0": {
        HVACMode.HEAT: "1",
        HVACMode.AUTO: "0",
    },
}

[climate.py.txt](https://github.com/user-attachments/files/18502963/climate.py.txt)

**Comment by donheinzo** [comment]
Same issue in 2026, still present in current v5.2.5. Root cause looks clear from the source (climate.py, HVAC_MODE_SETS["True/False"]): only HVACMode.HEAT: True is defined, there's no HVACMode.OFF: False entry. When the DP is False, nothing matches and the entity falls back to reporting auto — even though auto isn't in the configured hvac_modes list at all.

Setup: DP 1 (boolean power) used as HVAC Mode DP with "True/False" preset, on a Tuya electric floor heating thermostat (category wk). Toggling the DP true/false via the physical switch entity does not change the reported climate state away from auto.

Would a fix adding HVACMode.OFF: False to the "True/False" set (and equivalent updates to HVAC_ACTION_SETS where relevant) make sense, or is there a reason False was intentionally left unmapped?
